conan/tools/env/environment.py

```diff
@@ -520,12 +534,66 @@ def save_sh(self, file_location, generate_deactivate=True):
         content = f'script_folder="{os.path.abspath(filepath)}"\n' + content
         save(file_location, content)
 
+    def save_fish(self, file_location):
+        """Save a fish script file with the environment variables defined in the Environment object.
+
+        It generates a function to deactivate the environment variables configured in the Environment object.
+
+        :param file_location: The path to the file to save the fish script.
+        """
+        filepath, filename = os.path.split(file_location)
+        function_name = f"deactivate_{Path(filename).stem}"
```

Comment:
It follow the common name used by other virtualenv like: `conanbuild.sh` -> `deactivate_conanbuild.sh`. So you can actually can call from your terminal directly, like:

```
source conanbuild.fish
...
deactivate_conanbuild
```

In Fish, functions are exported as a command, so you can run directly.

About the random to avoid collision: In case we have multiple dependencies add bin folder to PATH, we will need to undo the configuration after deactivating it. Usually it would be store in a deactivate script, but here we are using a variable. Actually, the random is just an idea, but could be the build folder name too, because is kind unique name.

Reply:
Yes, having something deterministic would make more sense, I probably still didn't understand the issue.